Porting BT for windows

nk0leg opened the pull request:
see detain in SDLWIN-121

nk0leg commented:
@AGaliuzov, @pestOO, @anosach-luxoft ,  please review.

pestOO commented on CMakeLists.txt:
```diff
@@ -185,7 +186,11 @@ if(MSVC)
   if (${QT_PORT})
   # qt specific  flags
   else()
+<<<<<<< HEAD
```
legacy

pestOO commented on ms_tools/make_ssl_lib.cmd:
```diff
@@ -0,0 +1,21 @@
+git.exe clone https://github.com/openssl/openssl.git
+pop > NUL
+)
+IF NOT EXIST "C:\Build-OpenSSL-VC-64" (
```
hardcoded path

pestOO commented on src/3rd_party-static/libusbx-1.0.16/CMakeLists.txt:
```diff
@@ -23,11 +23,13 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
     ./libusb/os/threads_windows.c
     ./libusb/os/windows_usb.c
   )
+add_library("Libusb-1.0.16" ${LIBUSB_SOURCES})
```
duplicates at lime 41

pestOO commented on src/appMain/CMakeLists.txt:
```diff
@@ -134,8 +144,14 @@ include_directories (
   ${CMAKE_SOURCE_DIR}
   ${default_media_inc}
   ${LOG4CXX_INCLUDE_DIRECTORY}
+<<<<<<< HEAD
```
legacy

AGaliuzov commented on CMakeLists.txt:
```diff
@@ -185,7 +186,11 @@ if(MSVC)
   else()
+<<<<<<< HEAD
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /GF /GR /W3 /WX /Zc:wchar_t /favor:blend")# Add MSVS specific flags
+=======
```
Guys, I would suggest to run the code before pull requesting it in order to avoid such merge artifacts

pestOO commented on src/appMain/CMakeLists.txt:
```diff
@@ -150,6 +166,14 @@ set (HEADERS
 
+if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  list(APPEND LIBRARIES sqlite3 bthprops)
+#  message(STATUS "Additional lib path ${SQLITE_LIB_DIRECTORY} ${OPENSSL_LIB_DIRECTORY}")
```
legacy?

pestOO commented on src/components/hmi_message_handler/CMakeLists.txt:
```diff
@@ -60,12 +60,17 @@ set (SOURCES
     ${RTLIB}
 )
+else()
+set(LIBRARIES
```
idnent

pestOO commented on src/components/hmi_message_handler/CMakeLists.txt:
```diff
@@ -60,12 +60,17 @@ set (SOURCES
     ${DBUS_ADAPTER}
     ${RTLIB}
 )
+else()
```
May be split to set and append for each platform?

pestOO commented on src/components/media_manager/src/media_manager_impl.cc:
```diff
@@ -32,7 +32,7 @@
 #include "config_profile/profile.h"
 #include "media_manager/media_manager_impl.h"
-#include "media_manager/audio/from_mic_recorder_listener.h"
+//#include "media_manager/audio/from_mic_recorder_listener.h"
```
comment?

pestOO commented on src/components/media_manager/src/media_manager_impl.cc:
```diff
@@ -137,7 +137,7 @@ void MediaManagerImpl::StartMicrophoneRecording(
   const std::string& output_file,
   int32_t duration) {
-  LOG4CXX_INFO(logger_, "MediaManagerImpl::StartMicrophoneRecording to "
+  /*LOG4CXX_INFO(logger_, "MediaManagerImpl::StartMicrophoneRecording to "
```
TODO?

AGaliuzov commented on src/components/policy/src/policy/include/policy/pt_representation.h:
```diff
@@ -49,6 +49,7 @@ struct VehicleData {
   const std::string vehicle_make;
   const std::string vehicle_model;
   int vehicle_year;
+  VehicleData(){}
```
I guess compiler is able to generate default constructor, isn't it?

AGaliuzov commented on src/components/transport_manager/CMakeLists.txt:
```diff
@@ -99,12 +191,12 @@ if (BUILD_USB_SUPPORT)
-    ${TM_SRC_DIR}/usb/qnx/usb_connection.cc
-    ${TM_SRC_DIR}/usb/qnx/platform_usb_device.cc
-    )
+  # elseif(CMAKE_SYSTEM_NAME STREQUAL "QNX")
```
commented code? Do we really need it?

pestOO replied:
Do not delete QNX code please!

pestOO commented on src/appMain/main.cc:
```diff
@@ -88,6 +88,7 @@ bool InitHmi() {
   }
   return utils::System(kBrowser, kBrowserName).Add(kBrowserParams).Add(hmi_link)
       .Execute();
+  return true;
```
Oo

pestOO commented on src/components/include/utils/wsa_startup.h:
```diff
@@ -0,0 +1,51 @@
+  WsaStartup(BYTE minorVer, BYTE majorVer);
+  virtual ~WsaStartup();
+
+private:
```
legacy?

pestOO commented on src/components/transport_manager/CMakeLists.txt:
```diff
@@ -43,16 +44,107 @@ include_directories (
 )
 if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-    file (GLOB_RECURSE inc RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} include/transport_manager/*.h)
+  set (inc
```
Is it correct to include headers?

pestOO commented on src/components/transport_manager/src/bluetooth/bluetooth_device_scanner_win.cc:
```diff
@@ -0,0 +1,572 @@
+/*
+ * \file bluetooth_device_scanner.cc
```
Wrong license template

pestOO commented on src/components/transport_manager/src/tcp/tcp_client_listener.cc:
```diff
@@ -236,14 +279,14 @@ void TcpClientListener::StopLoop() {
   thread_stop_requested_ = true;
   // We need to connect to the listening socket to unblock accept() call
-  int byesocket = socket(AF_INET, SOCK_STREAM, 0);
+  int byesocket = (int)socket(AF_INET, SOCK_STREAM, 0);
```
old c-casting

pestOO commented on src/components/transport_manager/src/bluetooth/bluetooth_transport_adapter_win.cc:
```diff
@@ -0,0 +1,195 @@
+}
+
+bool BluetoothTransportAdapter::Restore() {
+  LOG4CXX_TRACE(logger_, "enter");
```
autotrace

pestOO commented on src/components/transport_manager/src/bluetooth/bluetooth_transport_adapter_win.cc:
```diff
@@ -0,0 +1,195 @@
+}
+
+void BluetoothTransportAdapter::Store() const {
+  LOG4CXX_TRACE(logger_, "enter");
```
autotrace

pestOO commented on src/components/transport_manager/src/bluetooth/bluetooth_transport_adapter_win.cc:
```diff
@@ -0,0 +1,195 @@
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "TransportManager")
+
+BluetoothTransportAdapter::~BluetoothTransportAdapter() {
```
redundant

pestOO commented on src/components/transport_manager/src/bluetooth/bluetooth_transport_adapter_win.cc:
```diff
@@ -0,0 +1,195 @@
+#include <sys/types.h>
+
+#ifdef OS_POSIX
+#include <sys/socket.h>
```
Do we need this and all other OS_posix if it is win only file?

pestOO commented on src/components/transport_manager/src/bluetooth/bluetooth_transport_adapter_win.cc:
```diff
@@ -0,0 +1,195 @@
+									  address,
+									  &addrSize);
+	if (ret_val != 0){
+		LOG4CXX_ERROR(logger_, "WSAAddressToString() failed with error code" << WSAGetLastError());
```
call cpplint please for this file

pestOO commented on src/components/transport_manager/src/tcp/tcp_client_listener.cc:
```diff
@@ -150,6 +163,32 @@ void SetKeepaliveOptions(const int fd) {
+  settings.onoff = 1;
+  settings.keepalivetime = keepidle * 1000;
+  settings.keepaliveinterval = keepintvl * 1000;
+//  setsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, (const char*)&yes, sizeof(yes));
```
legacy?

AGaliuzov commented on src/components/transport_manager/src/tcp/wsa_startup.cc:
```diff
@@ -0,0 +1,45 @@
+{
+  WSAData wsaData;
+  if (WSAStartup(MAKEWORD(majorVer, minorVer), &wsaData) != 0) {
+    throw;
```
as far as I know we don't use exceptions in SDL, isn't it?

Commit:
Fix Policy and TransportManager msvc build

Fix SecurityManager msvc build

SDLWIN-126

Fix copy_library_Policy build

SDLWIN-126

Porting BT for windows

see detain in SDLWIN-121

Fix libusb build for msvc

anosach-luxoft commented:
Pull request was resent (#13)
